sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -730,6 +730,7 @@ public Symbol visitMatchPredicate(MatchPredicate node, ExpressionAnalysisContext
             verifyTypesForMatch(identBoostMap.keySet(), columnType);
 
             Symbol queryTerm = castIfNeededOrFail(process(node.value(), context), columnType);
+            Preconditions.checkArgument(queryTerm instanceof Literal, SymbolFormatter.format("queryTerm must be a literal"));
```

Comment:
afaik the query term can actually be a subquery, in which case this check would fail but it's still valid.